l10n_ec_account_edi/models/account_edi_document.py

```diff
@@ -680,8 +683,43 @@ def _l10n_ec_get_info_debit_note(self):
             "importeTotal": self._l10n_ec_number_format(amount_total, 6),
             "pagos": debit_note._l10n_ec_get_payment_data(),
             "detalles": self._l10n_ec_header_get_document_lines_edi_data(taxes_data),
-            "infoAdicional": self._l10n_ec_get_info_aditional(),
+            "infoAdicional": self._l10n_ec_get_info_additional(),
         }
 
         debit_note_dict.update(self._l10n_ec_get_info_tributaria(debit_note))
         return debit_note_dict
+
+    @api.model
+    def send_mail_to_partners(self):
+        all_companies = self.env["res.company"].search(
+            [
+                ("partner_id.country_id.code", "=", "EC"),
+                ("l10n_ec_type_environment", "=", "production"),
+            ]
+        )
+        for company in all_companies:
+            self.with_company(company).send_mail_to_partner()
+
+    @api.model
+    def send_mail_to_partner(self):
+        account_moves = self.env["account.move"].search(
+            [
+                ("state", "=", "posted"),
+                ("is_move_sent", "=", False),
+                ("edi_document_ids.state", "=", "sent"),
+                ("partner_id.vat", "!=", "9999999999999"),
+            ]
+        )
+        for account_move in account_moves:
+            account_move.l10n_ec_send_email()
+
+        # Update documents with final consumer
+        account_moves_with_final_consumer = self.env["account.move"].search(
+            [
+                ("state", "=", "posted"),
+                ("is_move_sent", "=", False),
+                ("edi_document_ids.state", "=", "sent"),
+                ("partner_id.vat", "=", "9999999999999"),
```

Comment:
Final consumer can be a record with 13 or 10 digits, should be better to search for both cases

Reply:
done